homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -28,30 +26,38 @@ async def async_setup_entry(
     coordinator = hass.data[FRITZBOX_DOMAIN][entry.entry_id][CONF_COORDINATOR]
 
     for ain, device in coordinator.data.items():
-        if not device.has_alarm:
-            continue
-
-        entities.append(
-            FritzboxBinarySensor(
-                {
-                    ATTR_NAME: f"{device.name}",
-                    ATTR_ENTITY_ID: f"{device.ain}",
-                    ATTR_UNIT_OF_MEASUREMENT: None,
-                    ATTR_DEVICE_CLASS: DEVICE_CLASS_WINDOW,
-                    ATTR_STATE_CLASS: None,
-                },
-                coordinator,
-                ain,
-            )
-        )
+        for description in BINARY_SENSOR_DESCRIPTIONS:
+            if callable(description.suitable) and description.suitable(device):
+                entities.append(
+                    FritzboxBinarySensor(
+                        description,
+                        coordinator,
+                        ain,
+                    )
+                )
```

Comment:
even I would use a list comprehension for `for description in BINARY_SENSOR_DESCRIPTIONS`, loop `for ain, device in coordinator.data.items()` remains around, those it still needs the `entities.append` ... or do I miss something 🤔

Reply:
You can use multiple for's in a comprehension, something like
```py
FritzboxBinarySensor(...)
for ain, device in coordinator.data.items()
for description in BINARY_SENSOR_DESCRIPTIONS
if ...
```
Mix and match as you like. Btw this would be equivalent to
```py
for ain, device in coordinator.data.items():
    for description in BINARY_SENSOR_DESCRIPTIONS:
        if ...:
            FritzboxBinarySensor(...)
```